crates/sui-snapshot/src/lib.rs

```diff
@@ -240,5 +249,59 @@ pub async fn setup_db_state(
     committee_store.insert_new_committee(&next_epoch_committee)?;
     checkpoint_store.update_highest_executed_checkpoint(&last_checkpoint)?;
 
+    // TODO add progress bar
+    if verify {
+        eprintln!(
+            "Beginning DB live object state verification. This may take a while, \
+            and currently does not provide progress updates..."
+        );
+        // NB: this is technically incorrect as below we get the protocol config
```

Comment:
We know exactly at which epoch for each network when this is enabled. So it's probably not too bad if we just do a hardcoded check here to get the right simplified_unwrap_then_delete value?

Reply:
I was going back and forth on doing exactly this.. it would be invalidated if we wipe testnet... but if we do, the ChainId would also be invalidated, so I suppose we could do that as long as we have the right defaults

Reply:
Yeah that sounds reasonable (i.e. we will update this if chain id ever changes)

Reply:
Fixed with hard coded epochs. My epoch calculations are based on the following:

* `protocol_config.include_wrapped_tombstones` included starting at protocol verson 16
* Mainnet transitioned to version 16 on `2023-07-08` 
* Testnet transitioned to version 16 on `2023-06-29`
* Epoch on mainnet `2023-07-08`  went from 86 -> 87
* Epoch on testnet `2023-06-29`  went from 49 -> 50
* On the day that the epoch transitioned from `n - 1` to `n`, we started running the new protocol config, and at the end of day `n`, we created a snapshot for epoch `n` using the protocol config it was running at `n`
* Therefore if we restore to epoch `n` then the protocol config for epoch `n` dictates how that snapshot must be accumulated.

Unfortunately there's no easy way to test this to my knowledge - we don't maintain snapshots that are that old.

That said, because we don't maintain such old epochs, and the use case for restoring so far back is razor thin, I am not _too_ worried about it